packages/python-packages/api-stub-generator/apistub/nodes/_argtype.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+# Special default values that should not be treated as string literal
+SPECIAL_DEFAULT_VALUES = ["None", "..."]
+
+
+class ArgType:
+    """Represents Argument type
+    :param str: name
+        Name of the argument
+    :param str: argtype
+        Type of the argument. for e.g. str, int, BlobBlock
+    :param str: default
+        Default value for the argument, If any
+    """
+
+    def __init__(self, name, argtype=None, default=None):
+        super().__init__()
+        self.argname = name
+        self.argtype = argtype
+        self.default = default
+
+    def generate_tokens(
+        self, apiview, function_id, add_line_marker, include_default=True
+    ):
+        """Generates token for the node and it's children recursively and add it to apiview
+        :param ApiView: apiview
+        :param str: function_id
+            Module level Unique ID created for function 
+        :param bool: include_default
+            Optional flag to indicate to include/exclude default value in tokens
+        """
+        # Add arg name
+        id = None
+        if add_line_marker:
+            id = "{0}.param({1}".format(function_id, self.argname)
+            apiview.add_line_marker(id)
+
+        apiview.add_text(id, self.argname)
+        # add arg type
+        if self.argtype:
+            apiview.add_punctuation(":", False, True)
+            apiview.add_type(self.argtype)
+
+        # add arg default value
+        if include_default and self.default:
```

Comment:
Under what circumstances would `include_default` be true but `self.default` be empty? Or vice versa? If that happens should we add a warning?
Also, what if the default is `None`? In which case this statement will evaluate to `False`.

Reply:
I have removed include_default. It is not required anymore. `default` is set to `"None"` as per docstring or inspect returned value instead of `None`. inspect returns `Parameter.empty` when default is not set as opposed to default set to None